pkg/util/push/otel.go

```diff
@@ -69,6 +70,23 @@ func OTLPHandler(
 		}
 
 		reader := http.MaxBytesReader(nil, r.Body, int64(maxRecvMsgSize))
+		// Handle compression.
+		switch r.Header.Get("Content-Encoding") {
+		case "gzip":
+			gr, err := gzip.NewReader(reader)
+			if err != nil {
+				return nil, err
+			}
+			// Protect against compression blowing up too.
+			reader = http.MaxBytesReader(nil, gr, int64(maxRecvMsgSize))
```

Comment:
Can we initialize `reader` as `reader := r.Body` and wrap it into `http.MaxBytesReader` at once place only, not two (ie. `switch` statement handling compression).